docs/development/pigs/pig-012.rst

```diff
@@ -0,0 +1,340 @@
+
+    - Every tool in Gammapy knows how to serialise/read themselves
+    - Composite objects will be serialised as a series of all of their componets
+    - Serialisation will be a mix of YAML (models, state) and FITS files (maps)
+    - There will not be framework supporting different serialisation backends, we will have one way.
+
+**Session workflow**
+
+.. code::
+
+        $ mkdir analysis
+        $ cd analysis
+        $ edit gammapy_analysis_config.yaml
+        
+Then the user would type `ipython`, `juypter notebook` or write a script with the code below.
+
+.. code::
+
+        from gammapy.scripts import Analysis
+
+        analysis = Analysis(config)
+
+        analysis.select_observations()
+        # analysis.observations: gammapy.data.Observations
+
+        analysis.reduce_data()  # often slow, can be hours
+        # analysis.datasets: gammapy.datasets.Datasets
+
+        # If the user wants they can save all results from data reduction
+        # and re-start later. This stores config, datasets, ... all the
+        # analysis class state.
+        # analysis.write()
+        # analysis = Analysis.read())
+
+        analysis.optimise()  # often slow, can be hours
+
+        # Again, we could write and read, do the slow things only once.
+        # e.g. supervisor comes in and asks about significance of some
+        # model component or whatever
+
+        # Everything is accessible via the "analysis"
+        # It's like the Analysis in Fermipy or Sherpa "session" or "HESSArray" in HAP
+        # a global object that gives you access to everything.
+        # Method calls modify data members (e.g. models), but in between method calls
+        # advanced users can do a lot of custom processing.
+        # Many advanced use cases can be done with the Analysis API.
+        profile = analysis.model("source_42").spectrum.plot()
+
+        # Do we need energy_binning for the SED points in config or only here?
+        sed = analysis.spectral_points("source_42", energy_binning)
+
+**Commnad line tools**
+
+In addtition to ``gammapy analysis config``, we will have a ``gammapy analysis data_reduction`` 
+and a ``gammapy analysis optimise`` which perform the long processing tasks from the terminal 
+outside an ipython session or jupyter notebook, using all the information from the config 
+file and/or saved state.
+
+    - ``gammapy analysis config``: dumps a template configuration file
+    - ``gammapy analysis data_reduction``: performs a data reduction process
+    - ``gammapy analysis optimise``: performs a model fitting process
+ 
+ 
+Outlook
+=======
+
+Some of the use cases not covered by this high-level interface API are the following:
+
+    - Generation of simulated events and/or counts 
+    - Diagnostics (residuals, significance, TS)
+    - Iterative source detection methods
+    - Flexible background modelling
+    - Spectral points
```

Comment:
It will be easy to give spectral points via an Analysis method call, we shouldn't say here that it's not supported by the high-level API.

It is true that it's hard to configure spectral points via the config file, because one has to give a model name. But it is possible, if we want we can support spectral points via the config file.